modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -1552,18 +1552,30 @@ private boolean tryScrollOneCell(int targetIndex, boolean downOrRight) {
 
         T targetCell = getVisibleCell(targetIndex + indexDiff);
         if (targetCell != null) {
-            T cell = getAvailableCell(targetIndex);
-            setCellIndex(cell, targetIndex);
-            resizeCell(cell);
-            setMaxPrefBreadth(Math.max(getMaxPrefBreadth(), getCellBreadth(cell)));
-            cell.setVisible(true);
-            if (downOrRight) {
-                cells.addLast(cell);
-                scrollPixels(getCellLength(cell));
+            if (targetIndex < 0) {
+                T cell = getCell(targetIndex);
```

Comment:
Is there a reason you call `getCell` here, but `getAvailableCell` below?

Reply:
As I was also involved in fixing this error, I can answer.

The `getAvailableCell()` method creates a new cell if none exists for the specified index (-1). 
The `getCell()` returns the `accumCell` in this case, which is completely sufficient for our case, because we only need the dimensions of the cell here. The cell itself will not be added to the scene-graph.

Reply:
That sounds right.
@Maran23 does this answer your question?

Reply:
Yes. But is it possible to scroll to an index that is negative? (<0) And what do we estimate then?

Reply:
> But is it possible to scroll to an index that is negative?

Yes, both the negative value (``-1``) and the positive value, which is greater than the items count (to be precise ``last item index + 1``), are possible and indicate the direction in which the scrolling takes place.

The ``last item index + 1`` is already handled correctly. We only need to fix the handling of ``-1`` index.

Can you explain your second question in more detail?